main.tf

```diff
@@ -25,14 +25,14 @@ resource "aws_route" "public_internet_gateway" {
 resource "aws_route" "private_nat_gateway" {
   route_table_id         = "${element(aws_route_table.private.*.id, count.index)}"
   destination_cidr_block = "0.0.0.0/0"
-  nat_gateway_id         = "${element(aws_nat_gateway.natgw.*.id, count.index)}"
-  count                  = "${length(var.private_subnets) * lookup(map(var.enable_nat_gateway, 1), "true", 0)}"
+  nat_gateway_id         = "${element(aws_nat_gateway.natgw.*.id, count.index % length(var.azs))}"
```

Comment:
As a FYI, the `% length(var.azs)` in `"${element(aws_nat_gateway.natgw.*.id, count.index % length(var.azs))}"` is not necessary. 

Lists in HCL the [element function wraps](https://www.terraform.io/docs/configuration/interpolation.html#element-list-index-). This means that if you have a list with 3 items and you get the item with `index 3` will actually return the item with `index 0`
```python
test = ["foo", "bar", "baz"]
test[3]
> foo
```